dotcom-rendering/src/web/browser/sentryLoader/init.ts

```diff
@@ -16,7 +20,16 @@ const init = async (): Promise<void> => {
 	// Sentry 99% of the time. So instead we just do some basic math here
 	// and use that to prevent the Sentry script from ever loading.
 	const randomCentile = Math.floor(Math.random() * 100) + 1; // A number between 1 - 100
-	if (randomCentile <= 99 || sentryDisabled) {
+	// We want to log all errors for users in the bundle variant AB test
+	// Please ensure that the test sample rate is low
+	const isInBrowserVariantTest =
+		BUILD_VARIANT &&
+		window.guardian.config.tests[dcrJavascriptBundle('Variant')] ===
+			'variant';
+	const doNotLoadSentry =
+		(randomCentile <= 99 || sentryDisabled) && !isInBrowserVariantTest;
```

Comment:
I think this pattern is adding yet one more level of indirection around when to load Sentry, and I think refactoring by creating a pure function that determines wether to load it would be very beneficial. Reading this code, I’m struggling to parse it.

```ts
const shouldInjectSentry = (isDev: boolean, enableSentryReporting: boolean, isInBrowserVariantTest: boolean, centile: number): boolean => /* … */

// …

const init = () => {
	const skipSentryInjection = !shouldInjectSentry(isDev, enableSentryReporting, isInBrowserVariantTest, randomCentile);
	
	if (skipSentryInjection) {
		window.guardian.modules.sentry.reportError = (error) => {
			console.error(error);
		};
		return; // Do not initialise Sentry
	}
	
	/* … */
}
```

Reply:
I agree it is very confusing and in need of refactoring - the negative condition and flow is hard to read.

I was unsure if I should do a drive by refactoring as part of this ticket but I'm happy to do it now?

Reply:
I find it hard to resist drive-by refactors…

> If not now, when?
> If not you, who?

You could raise the refactor against this PR/branch to keep things as orthogonal as possible.